spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -220,14 +228,10 @@ public ThisT tableProperty(String name) {
     }
 
     protected T parse(Function<String, T> conversion, T defaultValue) {
-      if (!optionNames.isEmpty()) {
```

Comment:
The if here is redundant. We initially had `!= null` check before we added a list of options that is never null.